systemwidealerts.md

```diff
@@ -0,0 +1,80 @@
+# System-wide alerts
+[Back to the list of all defined endpoints](endpoints.md)
+
+This endpoint exposes system-wide alerts to be shown as banners in the frontend.
+Only administrators are allowed to create and edit these alerts, while all other users can view them.
+
+This feature will be expanded in the future
+* DSpace currently only supports a single system-wide alert, but the endpoints are designed to accomodate multiple alerts
+* Alerts can specify whether/how user sessions should be restricted while the alert is active, but this is a placeholder as session control has not been implemented yet.
+
+
+## Retrieve all alerts
+**GET /api/system/systemwidealerts**
+This endpoint returns a list of all system-wide alerts.
+
+Status codes:
+* 200 OK - if the operation succeeded
+
+
+## Retrieve a specific alert
+**GET /api/system/systemwidealerts/<:alert-id>**
+This endpoint returns a single system-wide alert by its integer identifier.
+
+Status codes:
+* 200 OK - if the operation succeeded 
+* 404 Not found - if no alert exists for the requested identifier
+
+
+## Create a new alert
+**POST /api/system/systemwidealerts**
+This endpoint allows administrators to create a new system-wide alert.
+
+```json
+{
+  "id": 1,
+  "message": "Alert message to show in the banner",
+  "allowSessions": 2,
+  "countdownTo": "2023-04-05T00:00:00.000Z",
+  "active": true
+}
+```
+
+The value of `countdownTo` must be a UTC datetime string.
+
+The possible values for `allowSessions` are
+* 0: allow all sessions
+* 1: allow current sessions only (admins will still be allowed to log in)
+* 2: allow admin sessions only (other users will be logged out)
```

Comment:
I'm not a fan of returning numbers and then having to describe their meaning.  It also then requires us to code the meaning any any client (for example our UI would need to know what these numbers mean). Why not return something that is more self explanatory?, e.g.
* "all" = "allow all sessions"
* "current" = "allow current sessions only"
* "admin" = "all admin sessions only"

For instance, we don't return the underlying numerical dspace object types from our REST API: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/core/Constants.java#L21-L41   Instead, the REST API returns "bistream", "bundle", "item", etc.

Reply:
Updated to reflect the string values you suggested, REST/Angular will be updated to match

I've kept the "explanations" in the contract though, so we won't need to look them up elsewhere later